tests/console/dns_srv.pm

```diff
@@ -36,7 +31,13 @@ sub run {
     systemctl 'show -p SubState named.service|grep SubState=running';
 
     # verify dns server responds to anything
-    assert_script_run "host localhost localhost";
+    my $e = script_run "host localhost localhost";
+    if (!$e) {
```

Comment:
I think you need to turn around the condition as script_run is returning not a true/false boolean value but the exit code, hence `if ($e) { … error handling … }`